eureka2-client/src/main/java/com/netflix/eureka2/client/interest/BatchingRegistryImpl.java

```diff
@@ -0,0 +1,170 @@
+                                }
+                                return null;
+                            }
+                        }).filter(new Func1<Boolean, Boolean>() {
+                            @Override
+                            public Boolean call(Boolean value) {
+                                return value != null;
+                            }
+                        }).materialize().flatMap(new Func1<Notification<Boolean>, Observable<Boolean>>() {
+                            @Override
+                            public Observable<Boolean> call(Notification<Boolean> materialized) {
+                                switch (materialized.getKind()) {
+                                    case OnNext:
+                                        return Observable.just(materialized.getValue());
+                                    case OnError:
+                                        if (logger.isDebugEnabled()) {
+                                            logger.debug("Swallowed observable error", materialized.getThrowable());
+                                        }
+                                }
+                                // Set all to FinishBuffering to close opened buffered streams
+                                for(Interest<T> i: interestsBufferingState.keySet()) {
+                                    interestsBufferingState.put(i, BufferingState.FinishBuffering);
+                                }
+                                return Observable.just(false);
+                            }
+                        }).doOnCompleted(new Action0() {
+                            @Override
+                            public void call() {
+                                interestsBufferingState.clear();
+                            }
+                        });
+                    }
+                }
+        ).subscribe(updatesSubject);
+    }
+
+    @Override
+    public void subscribe(Observable<ChangeNotification<T>> changeNotifications) {
+        notificationSources.onNext(changeNotifications);
+    }
+
+    @Override
+    public Observable<BufferingState> forInterest(final Interest<T> interest) {
+        return Observable.create(new OnSubscribe<BufferingState>() {
+            @Override
+            public void call(Subscriber<? super BufferingState> subscriber) {
+                updatesSubject.pause();
+                try {
+                    BufferingState current = shouldBatch(interest);
+                    if (current != BufferingState.FinishBuffering) {
+                        subscriber.onNext(current);
+                    }
+                    updatesSubject.map(new Func1<Boolean, BufferingState>() {
+                        @Override
+                        public BufferingState call(Boolean tick) {
+                            return shouldBatch(interest);
+                        }
+                    }).subscribe(subscriber);
+                } finally {
```

Comment:
RxJava will handle exceptions for us, if they happen. And in this code we do not expect any exception so providing custom ones is not needed.